src/main/java/com/google/sps/data/FakeAvailabilityDao.java

```diff
@@ -0,0 +1,137 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+/** Mimics accessing Datastore to support managing Availability entities. */
+public class FakeAvailabilityDao implements AvailabilityDao {
+  // @param datastore the fake database. Made public because it will only be used
+  // while testing.
+  public LinkedHashMap<Long, Availability> datastore;
+
+  /** Initializes the fields for FakeAvailabilityDao. */
+  public FakeAvailabilityDao() {
+    datastore = new LinkedHashMap<Long, Availability>();
+  }
+
+  /** Puts an Availability object into datastore with a randomly generated long as its id. */
+  @Override
+  public void create(Availability avail) {
+    long id = new Random().nextLong();
+    Availability toStoreAvail = avail.withId(id);
+    datastore.put(id, toStoreAvail);
+  }
+
+  /** Updates an Availability in datastore based on its id. */
+  @Override
+  public void update(Availability avail) {
+    datastore.put(avail.id(), avail);
+  }
+
+  /**
+   * Retrieves the Availability from datastore from the given id and wraps it in an Optional. If the
+   * Availability does not exist in datastore, the Optional is empty.
+   */
+  @Override
+  public Optional<Availability> get(long id) {
+    if (datastore.containsKey(id)) {
+      return Optional.of(datastore.get(id));
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Deletes all Availability entities for a user ranging from minTime to maxTime. minTime and
+   * maxTime are in milliseconds.
+   */
+  @Override
+  public void deleteInRangeForUser(String email, long minTime, long maxTime) {
+    List<Availability> userAvailability = getForUser(email);
+    List<Availability> userAvailabilityInRange = getInRange(userAvailability, minTime, maxTime);
+    for (Availability avail : userAvailabilityInRange) {
+      datastore.remove(avail.id());
+    }
+  }
+
+  /**
+   * Collects all Availabilities for the specified user within the specified time range, where
+   * minTime and maxTime are in milliseconds.
+   */
+  @Override
+  public List<Availability> getInRangeForUser(String email, long minTime, long maxTime) {
+    List<Availability> userAvailability = getForUser(email);
+    return getInRange(userAvailability, minTime, maxTime);
+  }
+
+  /**
+   * Collects all Availabilities for the specified user within the specified time range that have
+   * been scheduled over, where minTime and maxTime are in milliseconds.
+   */
+  @Override
+  public List<Availability> getScheduledInRangeForUser(String email, long minTime, long maxTime) {
```

Comment:
lf we decide to delete this in the DatastoreDao we should delete it here too.

Reply:
Deleted it in all instances (and relevant tests).